src/main/java/net/earthcomputer/clientcommands/command/PlayerInfoCommand.java

```diff
@@ -0,0 +1,106 @@
+    private static final Map<String, List<String>> cacheByUuid = new HashMap<>();
+
+    private static final JsonParser parser = new JsonParser();
+
+    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
+        addClientSideCommand("cplayerinfo");
+
+        LiteralCommandNode<ServerCommandSource> cplayerinfo = dispatcher.register(literal("cplayerinfo"));
+        dispatcher.register(literal("cplayerinfo")
+                .then(literal("namehistory")
+                        .then(argument("player", string())
+                                .suggests(((context, builder) -> suggestMatching(MinecraftClient.getInstance().world.getPlayers().stream().map(abstractPlayer -> abstractPlayer.getName().getString()), builder)))
+                                .executes(ctx -> getNameHistory(ctx.getSource(), getString(ctx, "player"))))));
+    }
+
+    private static int getNameHistory(ServerCommandSource source, String player) {
+        if (player.length() >= 32) {
+            if (cacheByUuid.containsKey(player)) {
+                sendFeedback(new TranslatableText("commands.cplayerinfo.getNameHistory.success", player, String.join(", ", cacheByUuid.get(player))));
+            } else {
+                fetchNameHistory(player);
+            }
+        } else {
+            if (cacheByName.containsKey(player)) {
+                sendFeedback(new TranslatableText("commands.cplayerinfo.getNameHistory.success", player, String.join(", ", cacheByName.get(player))));
+            } else {
+                Optional<String> optional = MinecraftClient.getInstance().world.getPlayers().stream()
+                        .filter(abstractPlayer -> abstractPlayer.getName().getString().equals(player))
+                        .map(Entity::getUuidAsString)
+                        .findFirst();
+                if (optional.isPresent()) {
+                    fetchNameHistory(optional.get());
+                } else {
+                    HttpClient client = HttpClient.newHttpClient();
+                    HttpRequest request = HttpRequest.newBuilder(URI.create("https://api.mojang.com/users/profiles/minecraft/" + player))
+                            .timeout(Duration.ofSeconds(5))
+                            .GET()
+                            .build();
+                    client.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+                            .thenApply(HttpResponse::body)
+                            .thenAccept(response -> {
+                                JsonElement result = parser.parse(response);
+                                if (result instanceof JsonNull) {
+                                    sendError(new TranslatableText("commands.cplayerinfo.ioException"));
+                                } else {
+                                    fetchNameHistory(result.getAsJsonObject().get("id").getAsString());
+                                }
```

Comment:
This will not run on the main thread, leading to thread safety issues when printing to chat. The body of this lambda needs to be wrapped in `MinecraftClient.getInstance().send` to schedule it to be run on the main thread. Same below